src/Microsoft.Fx.Portability/CompressedHttpClient.cs

```diff
@@ -93,47 +106,118 @@ public CompressedHttpClient(ProductInformation info)
                 FileExtension = ".json"
             }; ;
 
-            var response = await CallInternalAsync(request, json);
-            var result = response.Response.Deserialize<TResponse>();
+            var response = await CallInternalAsync(request, new[] { json });
+            var result = response.Response.FirstOrDefault().Data.Deserialize<TResponse>();
```

Comment:
I've switched this to just use `.Single()` since there should be 1 and only 1 in the enumerable. This is because the length of the array passed to `CallInternalAsync` should be the same as the output. The `OrDefault` ones should be used if there's a chance it will be an empty enumerable; that is not the case with this.